src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -542,14 +542,13 @@ public String Insert(int startIndex, String value)
             return result;
         }
 
-        public static string Join(char separator, params string[] value)
+        public unsafe static string Join(char separator, params string[] value)
         {
             if (value == null)
             {
                 throw new ArgumentNullException(nameof(value));
             }
-
-            return Join(separator, value, 0, value.Length);
+            return JoinCore(&separator, 1, value, 0, value.Length);
```

Comment:
@AlexRadch I am unsure if changing the code like this will make things faster. When the code is run, the JIT compiler should inline this `Join` call, so that it would be effectively the same as writing `JoinCore(&separator, 1, value, 0, value.Length)`.